added annotation steps and job

jmmut opened the pull request:
Now, the main job has 3 extra steps to annotate the vcf. They are skippable and launchable in isolation as a standalone job. The steps are:
- annotPreCreate: dump the list of variants in a VEP-readable format.
- annotCreate: getRuntime().exec("perl vep.pl ...")
- annotLoad: load the output of VEP into the DB.

jmmut committed:
pipeline: annot step 1/3: pre create.

this step creates a list of variants to annotate.
that file will be feeded to VEP.

jmmut committed:
pipeline: annot step 2/3: create annotation.

implemented with a external process with Runtime.exec(), and writing to/reading
from its streams.

jmmut committed:
pipeline: pre create annotation: small improvement moving data

jmmut committed:
pipeline: annot step 3/3: added test

jmmut committed:
pipeline: annot step 3/3: implemented

jmmut committed:
pipeline: annotation: splitting vep parameters

jmmut committed:
pipeline: updating launch script

jmmut committed:
restored DB clean

jmmut committed:
pipeline: improved documentation, mainly README

jmmut committed:
pipeline: added annotation standalone job

cyenyxe commented on eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/jobs/VariantAnnotConfiguration.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    @Bean
+    public Job variantStatsJob() {
```
Should be "variantAnnotJob"

cyenyxe commented on eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/jobs/VariantAnnotConfiguration.java:
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 EMBL - European Bioinformatics Institute
```
We are in 2016, time goes fast! :smile:

jmmut replied:
Do we have to change the year of all the files?

cyenyxe replied:
Only for the files that have been created/modified in 2016. So it could not apply exactly to the file I have highlighted, but to others.

If there were created in 2015, then the copyright declaration should read "2015-2016", and if they were created this year, just "2016".

cyenyxe commented on eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotPreCreate.java:
```diff
@@ -0,0 +1,100 @@
+ *
+ * @author Jose Miguel Mut Lopez &lt;jmmut@ebi.ac.uk&gt;
+ */
+public class VariantsAnnotPreCreate implements Tasklet {
```
The class name does not describe what it does. Maybe it could be something like "VariantsAnnotGenerateInput" "VariantsAnnotInputGeneration"?

jmmut committed:
pipeline: naming polishments

- renaming VariantsAnnotPreCreate to VariantsAnnotGenerateInput
- updating copyright year
- correct name of VariantAnnotJob

jmmut commented:
I'm fixing a bug regarding ensembl coordinates, waiting to use https://github.com/opencb/biodata/pull/88

jmmut committed:
pipeline: generateInput now uses biodata's copyInEnsemblFormat

jmmut committed:
using biodata 0.4.5 and removing duplicate dependencies
